### PR TITLE
Added broadcast

### PR DESCRIPTION
**Pull request by Accesse**
Included broadcast in WebSocketServer.js

### CODE REVIEW

File: lib/WebSocketServer.js [hunk 1]
```diff
@@ -102,6 +102,26 @@ function WebSocketServer(options, callback) {
...
+ 
+ WebSocketServer.prototype.broadcast = function(msg) {
+  // broadcast to all clients
+  var error = null;
+  try {
+    for (var i = 0, l = this.clients.length; i < l; ++i) {
+      this.clients[i].send(msg);
+    }
+  }
+  catch (e) {
```
Comment (JacksonTian) [inline lib/WebSocketServer.js]: one error will break other clients

Comment (Accesse) [inline lib/WebSocketServer.js]: @JacksonTian That's right. what if we remove the try/catch?

Comment (JacksonTian) [inline lib/WebSocketServer.js]: try catch the `this.clients[i].send(msg);`

Comment (Accesse) [inline lib/WebSocketServer.js]: @JacksonTian  Like that?
```
WebSocketServer.prototype.broadcast = function(msg) {
 // broadcast to all clients
 var error = null;
   
   for (var i = 0, l = this.clients.length; i < l; ++i) {
       try {
           this.clients[i].send(msg);
       }
       catch (e) {
              error = e;
       }
   }

}
```

File: lib/WebSocketServer.js [hunk 2]
```diff
@@ -102,6 +102,27 @@ function WebSocketServer(options, callback) {
...
+  var error = null;
+  for (var i = 0, l = this.clients.length; i < l; ++i) {
+    try {
+      this.clients[i].send(msg);
+    }
+    catch (e) {
+      error = e;
+    }
+  }
+
```
Comment (JacksonTian) [inline lib/WebSocketServer.js]: how do you process the error?

**Comment by 3y3** [comment]
Why we expect this in `ws` lib?
There is no access to private fields, so each user can implement this by himself.
`ws` should not to be `socket.io`

**Comment by lpinca** [comment]
I agree with @3y3.

**Comment by 3rd-Eden** [comment]
This does not belong in a core library like ws, there are 1000's of ways to implement broadcasting and I don't want to force this implementation or any other upon users. The README already includes a simple broadcasting example as well.

**Pull request closed without merging**